vendor/k8s.io/kubernetes/pkg/securitycontextconstraints/capabilities/mustrunas.go

```diff
@@ -101,9 +101,21 @@ func (s *defaultCapabilities) Validate(pod *api.Pod, container *api.Container) f
 		return allErrs
 	}
 
+	allowedAdd := makeCapSet(s.allowedCaps)
+	allowAllCaps := allowedAdd.Has(string(api.CapabilityAll))
+	if allowAllCaps {
+		if len(s.requiredDropCapabilities) > 0 {
```

Comment:
I've pushed the code that is addressing your first comment.

>Do we need a migration of an invalid strategy?

While technically it was possible to create a SCC with `*` as the `allowedCapabilities` value, most likely no one uses it because it didn't have a special meaning.